akka-stream/src/main/scala/akka/stream/FlowMonitor.scala

```diff
@@ -60,5 +60,5 @@ object FlowMonitorState {
   /**
    * Java API
    */
-  def finished[U]() = Finished
+  def finished[U](): StreamState[U] = Finished
```

Comment:
>  I think the bincomp can be ignored since we are just making the return type more generic.

@johanandren it's not source compatible:

```
FlowMonitorState$Finished fin = FlowMonitorState.finished()
```

However, I think we can make an exception and allow this change. Seems unlikely that it will cause any trouble.

BTW, trait + object is used here so it will not work with java when using the Scala 2.11 version. (nothing we need to change though)